packages/core/upload/admin/src/hooks/useBulkRemove.js

```diff
@@ -0,0 +1,60 @@
+import { useMutation, useQueryClient } from 'react-query';
+import { useNotification } from '@strapi/helper-plugin';
+
+import pluginId from '../pluginId';
+import { axiosInstance, getRequestUrl } from '../utils';
+
+export const useBulkRemove = () => {
+  const toggleNotification = useNotification();
+  const queryClient = useQueryClient();
+  const url = getRequestUrl('actions/bulk-delete');
+
+  const bulkRemoveQuery = filesAndFolders => {
+    const payload = filesAndFolders.reduce((acc, selected) => {
+      const { id, type } = selected;
+      const key = type === 'asset' ? 'fileIds' : 'folderIds';
+
+      if (!acc[key]) {
+        acc[key] = [];
+      }
+
+      acc[key].push(id);
+
+      return acc;
+    }, {});
+
+    return axiosInstance.post(url, payload);
+  };
+
+  const mutation = useMutation(bulkRemoveQuery, {
+    onSuccess: res => {
+      const {
+        data: { data },
+      } = res;
+
+      if (data?.files?.length > 0) {
+        queryClient.refetchQueries([pluginId, 'assets'], { active: true });
+        queryClient.refetchQueries([pluginId, 'asset-count'], { active: true });
+      }
+
+      if (data?.folders?.length > 0) {
+        queryClient.refetchQueries([pluginId, 'folders'], { active: true });
+      }
+
+      toggleNotification({
+        type: 'success',
+        message: {
+          id: 'modal.remove.success-label',
+          defaultMessage: 'The folder has been successfully removed.',
+        },
+      });
+    },
+    onError: error => {
+      toggleNotification({ type: 'warning', message: error.message });
+    },
+  });
+
+  const remove = (...args) => mutation.mutateAsync(...args);
+
+  return { ...mutation, remove };
+};
```

Comment:
Do I need to test also the bulk remove? It seems to be broken, especially the select step
![bulk](https://user-images.githubusercontent.com/7756284/168388782-82d9cdf1-1884-4f47-b70c-2ccbbc960314.gif)

Reply:
I think these problems would go away after you re-compile the helper-plugin based on that branch. I did some updates there and it looks like you are missing them. Could you give it a try? I think bulk remove should be ready with this branch ...

Reply:
Indeed 😅